@here/harp-mapview/lib/geometry/PhasedTileGeometryLoader.ts

```diff
@@ -0,0 +1,420 @@
+        }
+
+        return this.m_currentPhaseIndex < this.m_loadPhaseDefinitions.length
+            ? this.m_currentPhaseIndex
+            : undefined;
+    }
+
+    /**
+     * Create all geometries of the specified [[GeometryKind]] `kindToCreate`.
+     *
+     * @param {TileGeometryCreator} geometryCreator
+     * @param {GeometryKind} kindToCreate
+     */
+    protected createKind(geometryCreator: TileGeometryCreator, kindToCreate: GeometryKind): void {
+        if (this.m_geometryKindsLoaded.has(kindToCreate)) {
+            return;
+        }
+        this.m_geometryKindsLoaded.add(kindToCreate);
+
+        const tile = this.tile;
+        const decodedTile = this.m_decodedTile;
+
+        if (decodedTile !== undefined) {
+            if (!tile.hasGeometry && tile.dataSource.addTileBackground) {
+                geometryCreator.addGroundPlane(tile);
+            }
+
+            const filter = (technique: Technique): boolean => {
+                if (technique.enabled !== true) {
+                    return false;
+                }
+
+                const techniqueKind = technique.kind;
+
+                // All kinds are allowed, except those which are explicitly disabled.
+                if (kindToCreate === GeometryKind.All) {
+                    return true;
+                }
+
+                if (techniqueKind instanceof Set) {
+                    const techniqueKinds = techniqueKind as GeometryKindSet;
+
+                    // Check if that technique fits the expected kindToCreate.
+                    return techniqueKinds.has(kindToCreate);
+                } else {
+                    return techniqueKind === kindToCreate;
+                }
+            };
+
+            geometryCreator.createObjects(tile, decodedTile, filter);
+
+            const textFilter = (technique: Technique): boolean => {
+                if (
+                    !isPoiTechnique(technique) &&
+                    !isLineMarkerTechnique(technique) &&
+                    !isTextTechnique(technique)
+                ) {
+                    return false;
+                }
+                return filter(technique);
+            };
+
+            // TextElements do not get their geometry created by Tile, but are managed on a
+            // higher level.
+            geometryCreator.createTextElements(tile, decodedTile, textFilter);
+
+            geometryCreator.preparePois(tile, decodedTile);
+        }
+    }
+
+    protected processTechniques(
+        enabledKinds: GeometryKindSet | undefined,
+        disabledKinds: GeometryKindSet | undefined
+    ): void {
+        const decodedTile = this.m_decodedTile;
+
+        if (decodedTile === undefined) {
+            return;
+        }
+
+        for (const technique of decodedTile.techniques) {
+            // Make sure that all technique have their geometryKind set, either from the Theme or
+            // their default value.
+            let geometryKind = technique.kind;
```

Comment:
Do we even need this variable? Why not just check/assign technique.kind directly?